### PR TITLE
Migrate from Yarn v1 to Yarn v3

### LINKED ISSUE #847: Migrate to Yarn 3
We want to migrate from Yarn v1 to Yarn v3. The main reason for migrating is that Yarn v3 is better maintained. Yarn v1 is not being maintained anymore, and includes various bugs that will never be fixed. We're also hoping that certain Yarn v3 will be useful when this repository is converted into a monorepo.

### PR DESCRIPTION
**Pull request by Gudahtt**
The package manager used has been upgraded from Yarn v1 to Yarn v3. All of the steps in the [migration guide](https://yarnpkg.com/getting-started/migration) have been followed. Effectively everything should work the same way it did before.

Closes #847

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -259,9 +259,8 @@ console.log(datamodel.flatState); // {infura: {...}, contractExchangeRates: [...
 
 - Install [Node.js](https://nodejs.org) version 12
   - If you are using [nvm](https://github.com/creationix/nvm#installation) (recommended) running `nvm use` will automatically choose the right node version for you.
-- Install [Yarn v1](https://yarnpkg.com/en/docs/install)
-- Run `yarn setup` to install dependencies and run any requried post-install scripts
-  - **Warning:** Do not use the `yarn` / `yarn install` command directly. Use `yarn setup` instead. The normal install command will skip required post-install scripts, leaving your development environment in an invalid state.
+- Install [Yarn v3](https://yarnpkg.com/getting-started/install)
+- Run `yarn install` to install dependencies and run any requried post-install scripts
```
Comment (mcmire) [inline README.md:263]: typo: `requried` -> `required`

Comment (Gudahtt) [reply README.md:263]:
```suggestion
- Run `yarn install` to install dependencies and run any required post-install scripts
```

**Review by mcmire — APPROVED** [review]
Looks good!

File: .yarnrc.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# We use allow-scripts to manually enable scripts and then run them
```
Comment (mcmire) [inline .yarnrc.yml:1]: I've found that this file will get regenerated if you add a new plugin. So that's one thing to know about this file :(

**Review by mcmire — COMMENTED** [review]
Noted a thing I found while working with the monorepo POC branch. I also discovered that Prettier will attempt to lint `.yarnrc.yml`. Should we make this change in `package.json`?

``` diff
- "lint:misc": "prettier '**/*.json' '**/*.md' '!CHANGELOG.md' '**/*.yml' --ignore-path .gitignore",
+ "lint:misc": "prettier '**/*.json' '**/*.md' '!CHANGELOG.md' '**/*.yml' '!.yarnrc.yml' --ignore-path .gitignore",
```

File: .yarnrc.yml (hunk 2 above)
Comment (Gudahtt) [reply .yarnrc.yml:1]: I had this problem as well. I was preserving them manually. Pretty inconvenient.

**Comment by Gudahtt** [comment]
>I also discovered that Prettier will attempt to lint .yarnrc.yml. Should we make this change in package.json?

Good idea! Done.

**Review by mcmire — APPROVED** [review]
LGTM!

**Pull request merged**